src/Type/ObjectManagerDynamicReturnTypeExtension.php

```diff
@@ -11,19 +11,24 @@
 use PHPStan\Type\ObjectType;
 use PHPStan\Type\Type;
 
+/**
+ * @deprecated This class will be dropped once support for TYPO3 <= 10 is dropped.
+ */
 class ObjectManagerDynamicReturnTypeExtension implements DynamicMethodReturnTypeExtension
 {
 
 	public function getClass(): string
 	{
-		return \TYPO3\CMS\Extbase\Object\ObjectManagerInterface::class;
+		return interface_exists(\TYPO3\CMS\Extbase\Object\ObjectManagerInterface::class)
+			? \TYPO3\CMS\Extbase\Object\ObjectManagerInterface::class : '';
```

Comment:
Do we really need this?

Reply:
Yes, or something else, because the interface is gone in TYPO3 12, and PHPStan will complain about the missing interface.

What else could we do about this?